code/game/objects/items/cigs_lighters.dm

```diff
@@ -227,7 +227,10 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		return
 	open_flame()
 	if(reagents && reagents.total_volume)
-		handle_reagents()
+		nextdragtime++

```

Comment:
Instead of this, take a look at how `world.time` is used for cooldowns, some examples:
https://github.com/tgstation/tgstation/blob/1b50671ca072dd646e98bee971a6d20d5c06a2de/code/modules/mining/equipment/mineral_scanner.dm#L19-L21
https://github.com/tgstation/tgstation/blob/608353fe3b7b2039c2faf1ec34805e86ee1078a4/code/modules/cargo/bounty_console.dm#L82-L84

Reply:
Aight, will take a look later today.

Reply:
ok I did the thing